tests/unit/src/test/scala/tests/DefinitionLspSuite.scala

```diff
@@ -723,10 +724,17 @@ class DefinitionLspSuite
         workspace,
       )
       _ = assert(locations.length == 1)
-      _ = assert(locations.forall(_.getUri().endsWith("a/Other.scala")))
+      (expectedFile, expectedLine) =
+        if (V.latestScala3Next.startsWith("3.6")) {
+          // We changed order to first look for definition, but there is a performance bug fixed in 3.7.0
```

Comment:
I think this bug might need to be looked into again... I'm not sure that will all the changes in go to definition it won't resurface for `3.7.x` too. And we had a workaround that made it work for `3.4.x - 3.6.x`, when semanticdb was used as the first strategy (https://github.com/scalameta/metals/pull/6344).

Reply:
Checked that it works for `3.7.x`.